docs/tutorials/tut-design.md

```diff
@@ -0,0 +1,369 @@
+ - **url**: the URL Cortex XSOAR should connect to. In case the API is SaaS and the URL is the same for all customers and is not going to change over time, you can omit this parameter and store the URL directly in the code. Some SaaS products need just the name of the tenant instead of the full URL (as they add the common suffix in the code). For example: `customer1.my-test-saas-service.info` and `customer2.my-test-sas-service.info`: in this case the input parameter could just be the name of the tenant (`customer1` or `customer2`).
+
+ - **api_key**: usually credentials are needed to authenticate to the API you want to interact with. Many times this is in the form of an API Key, a secure quantity that gets stored in Cortex XSOAR and used in somewhere in your code (i.e. to populate HTTP headers) to craft the requests. Sometimes APIs use different authentication parameters: [credentials](../reference/articles/managing-credentials), tokens, client_id and client_secret combination, and so on. Make sure you capture all the required parameters that will allow a machine-to-machine authentication between Cortex XSOAR and your API.
+ 
+    **Note**: if you are using authentication mechanisms based on short-lived access tokens such as JSON Web Tokens (JWT) and long-lived refresh tokens, take a look at the [Integration Cache](../integrations/integration-cache) article.
+
+Every integration also requires some additional parameters:
+- **proxy**: boolean to determine whether to use the system proxy setting. You won't actually need to deal with proxy settings in your code, just set environment variables accordingly based on this boolean.
+
+- **insecure**: boolean to determine whether to avoid verifying SSL certificates. This parameter, which by default is set to False, is important when customers deploy Cortex XSOAR on-prem behind proxies with self-signed certificates.
+
+
+Integration parameters should be named using [snake_case](https://medium.com/better-programming/string-case-styles-camel-pascal-snake-and-kebab-case-981407998841).
+
+#### Fetch Incidents Parameters
+
+If your integration [fetches incidents](../integrations/fetching-incidents), you should define what type of entities in the third party product you are connecting to you are retrieving. Every product has its own nomenclature: they can be called Alerts, Incidents, etc: for the rest of this section we are going to assume they are called Alerts in the third party product, and they are mapped 1:1 to Cortex XSOAR incidents.
+
+Many products are very verbose, and can potentially generate lots of alerts of different types, with different level of severity. Each alert could have a status (i.e. `open` or `resolved`) as well as attributes.
+
+SOCs analysts might be interested only in a subset of the incoming Alerts, so when they configure your integration in Cortex XSOAR, they expect to find Parameters that allow them to filter and determine what alerts are going to generate incidents or discarded.

```

Comment:
i would also add a note about a free query string to filter incidents if the product allows it. 
users are used to the syntax and it allows freedom to search exact incidens